src/psyclone/tests/domain/lfric/lfric_builtins_test.py

```diff
@@ -407,38 +407,38 @@ def test_get_argument_symbols(monkeypatch):
     psy = PSyFactory(API, distributed_memory=False).create(invoke_info)
     loop = psy.invokes.invoke_list[0].schedule[0]
     kern = loop.loop_body[0]
-    symbols = kern.get_argument_symbols()
-    for sym in symbols:
-        assert isinstance(sym, DataSymbol)
-        if sym.name == "a":
-            # We should have a symbol of ScalarType for the scalar
-            assert isinstance(sym.datatype, ScalarType)
-            assert sym.datatype.intrinsic == ScalarType.Intrinsic.REAL
-        else:
-            assert isinstance(sym.datatype, DataTypeSymbol)
-            assert sym.datatype.name == "field_proxy_type"
-    # Repeat but force the symbols for the scalar and field-proxy args to be
-    # created
-    del loop.parent.symbol_table._symbols["a"]
-    del kern.parent.symbol_table._symbols["f2_proxy"]
-    symbols = kern.get_argument_symbols()
-    for sym in symbols:
-        if sym.name == "a":
-            # We should have a symbol of ScalarType for the scalar
-            assert isinstance(sym.datatype, ScalarType)
-            assert sym.datatype.intrinsic == ScalarType.Intrinsic.REAL
-        else:
-            assert isinstance(sym.datatype, DataTypeSymbol)
-            assert sym.datatype.name == "field_proxy_type"
-
-    # Monkeypatch the scalar argument so that it appears to be an operator.
-    monkeypatch.setattr(kern._arguments.args[1], "_argument_type",
-                        LFRicConstants().VALID_OPERATOR_NAMES[0])
-    with pytest.raises(NotImplementedError) as err:
-        kern.get_argument_symbols()
-    assert ("Unsupported Builtin argument type: 'a' is of type "
-            "'{0}'".format(LFRicConstants().VALID_OPERATOR_NAMES[0]) in
-            str(err.value))
+    refs = kern.get_indexed_field_argument_references()
```

Comment:
Can you assert that the `len(refs)` is the expected one for this kern call.

Reply:
Done.